packages/slice-machine/components/AppLayout/Navigation/index.test.tsx

```diff
@@ -133,6 +156,95 @@ describe("Side Navigation", () => {
     }
   );
 
+  test("should display the number of changes on the 'Changes' item", async () => {
+    renderSideNavigation({ canUpdate: true, hasChanges: true });
+
+    expect(await screen.findByText("Changes")).toBeVisible();
+    const changesItem = screen
+      .getByText("Changes")
+      .closest("a") as HTMLAnchorElement;
+
+    expect(within(changesItem).getByText("1")).toBeVisible();
+  });
+
+  test("should not display the number of changes or 'Disconnect' on the 'Changes' item", async () => {
+    renderSideNavigation({ canUpdate: true, hasChanges: false });
+
+    expect(await screen.findByText("Changes")).toBeVisible();
+    const changesItem = screen
+      .getByText("Changes")
+      .closest("a") as HTMLAnchorElement;
+
+    expect(within(changesItem).queryByText("1")).not.toBeInTheDocument();
+    expect(
+      within(changesItem).queryByText("Login required")
+    ).not.toBeInTheDocument();
+  });
+
+  test("should display the information that user is disconnected on the 'Changes' item when unauthorized", async () => {
+    renderSideNavigation({
+      canUpdate: true,
+      hasChanges: true,
+      authStatus: AuthStatus.UNAUTHORIZED,
+    });
+
+    expect(await screen.findByText("Changes")).toBeVisible();
+    const changesItem = screen
+      .getByText("Changes")
+      .closest("a") as HTMLAnchorElement;
+
+    expect(within(changesItem).getByText("Login required")).toBeVisible();
+  });
+
+  test("should display the information that user is disconnected on the 'Changes' item when forbidden", async () => {
+    renderSideNavigation({
+      canUpdate: true,
+      hasChanges: true,
+      authStatus: AuthStatus.FORBIDDEN,
+    });
+
+    expect(await screen.findByText("Changes")).toBeVisible();
+    const changesItem = screen
+      .getByText("Changes")
+      .closest("a") as HTMLAnchorElement;
+
+    expect(within(changesItem).getByText("Login required")).toBeVisible();
+  });
+
+  test("should display the information that user is disconnected on the 'Changes' item when offline", async () => {
+    vi.spyOn(navigator, "onLine", "get").mockReturnValueOnce(false);
```

Comment:
hmm, that's an odd edge case, as they could have a valid login token but there's no why to check it

Reply:
True, but I did the same stuff as we are doing in the Changes page, so if it's offline you cannot see the page.